plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/browser/BrowserPopup.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2023 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.ui.browser;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.SWTError;
+import org.eclipse.swt.browser.Browser;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Shell;
+import org.jkiss.code.NotNull;
+import org.jkiss.dbeaver.Log;
+import org.jkiss.dbeaver.ui.UIUtils;
+import org.jkiss.dbeaver.ui.dialogs.AbstractPopupPanel;
+
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+public class BrowserPopup extends AbstractPopupPanel {
+    private static final Log log = Log.getLog(BrowserPopup.class);
+    private static final Map<String, Browser> browserRegistry = new HashMap<>();
+
+    private final String id;
+    private final URL url;
+
+    public static void openBrowser(@NotNull String id, @NotNull URL url) {
+        Browser browser = browserRegistry.get(id);
+        if (browser != null && !browser.isDisposed()) {
+            browser.setUrl(url.toString());
+        } else {
+            BrowserPopup browserPopup = new BrowserPopup(UIUtils.getActiveWorkbenchShell(), id, url);
+            browserPopup.open();
+        }
+    }
+
+    @Override
+    protected Composite createDialogArea(Composite parent) {
+        Composite composite = UIUtils.createComposite(parent, 1);
+        GridData gd = new GridData(GridData.FILL_BOTH);
+        gd.heightHint = 500;
+        gd.widthHint = 800;
+        composite.setLayoutData(gd);
+        try {
+            Browser browser = new Browser(composite, SWT.NONE);
+            gd = new GridData(GridData.FILL_BOTH);
+            gd.widthHint = 600;
+            gd.heightHint = 500;
+            browser.setLayoutData(gd);
+            browserRegistry.put(id, browser);
+            browser.setUrl(url.toString());
+        } catch (SWTError e) {
+            log.error("Could not instantiate Browser", e);
```

Comment:
It should be a fatal error IMO

Reply:
There are 100 reasons why it will go wrong, especially on linux, in that case we warn the user and open using external browser